File: BEECROWD/2395_navio_apisc2.js


/*
1 1 1
1 1 1

1 2 5
9 6 11

1 2 12
6 9 10
*/

var input = require('fs').readFileSync('./dev/stdin', 'utf8')
var lines = input.split('\r').join('').split('\n')

function main() {
  const container_atribs = cada_indice_inteiro(lines.shift().split(' '))
  const navio_atribs = cada_indice_inteiro(lines.shift().split(' '))

  if (
    eh_maior_igual(navio_atribs[0], container_atribs[0]) &&
    eh_maior_igual(navio_atribs[1], container_atribs[1]) &&
    eh_maior_igual(navio_atribs[2], container_atribs[2]) 
  ) {
    const qtd_containers = calcular_proporcao(navio_atribs, container_atribs)
    exibir(qtd_containers)
  } else {
    exibir(0)
  }
}

function exibir(conteudo) {
  console.log(conteudo)
}

function tornar_inteiro(alvo) {
  return parseInt(alvo)
}

function cada_indice_inteiro(iteravel) {
  lista_de_inteiros = []

  for (let i = 0; i < iteravel.length; i++) {
    if (iteravel[i] != ' ') {
      lista_de_inteiros.push(tornar_inteiro(iteravel[i]))
    }
  }

  return lista_de_inteiros
}

function eh_maior_igual(alvo, ref) {
  return alvo >= ref
}

function calculo_preciso(alvo, ref) {
  return Math.floor(alvo / ref)
}

function calcular_proporcao(alvo, ref) {
  let qtd_containers = 1
  
  for (let i = 0; i < alvo.length; i++) {
    qtd_containers *= (calculo_preciso(alvo[i], ref[i]))
  }
  return qtd_containers
}

main()

File: BEECROWD/2395_navio_apisc2.py


class Ship:

    def __init__(self):
        self.container = self.entrada_txt()
        self.navio = self.entrada_txt()
        self.container_int = self.cada_indice_inteiro(self.container)
        self.navio_int = self.cada_indice_inteiro(self.navio)

        if self.eh_maior_igual(self.navio_int[0], self.container_int[0]) \
           and self.eh_maior_igual(self.navio_int[1], self.container_int[1]) \
           and self.eh_maior_igual(self.navio_int[2], self.container_int[2]):

            self.qtd_container = self.calcular_proporcao(self.navio_int, self.container_int)
            print(self.qtd_container)
        else:
            print(0)

    @staticmethod
    def entrada_txt():
        entrada = input().split(' ')
        return entrada

    @staticmethod
    def cada_indice_inteiro(iteravel):
        novo = []
        for i in iteravel:
            novo.append(int(i))
        return novo

    @staticmethod
    def eh_maior_igual(alvo, ref):
        return alvo >= ref

    @staticmethod
    def calcular_proporcao(a, b):
        novo = []

        calculo = 1
        for i in range(len(a)):
            novo.append(a[i] // b[i])

        for i in range(len(novo)):
            calculo *= novo[i]

        return calculo


ship = Ship()
